core/field_dropdown.ts

```diff
@@ -744,13 +669,89 @@ const IMAGE_Y_PADDING: number = IMAGE_Y_OFFSET * 2;
 /** Android can't (in 2014) display "▾", so use "▼" instead. */
 FieldDropdown.ARROW_CHAR = userAgent.ANDROID ? '▼' : '▾';
 
+/**
+ * NOTE: Because Sentinel is an empty class, proving a value is Sentinel does
+ * not resolve in TS that it isn't a MenuGenerator.
+ */
+function isMenuGenerator(menuGenerator: MenuGenerator|
+                         Sentinel): menuGenerator is MenuGenerator {
+  return menuGenerator !== Field.SKIP_SETUP;
+}
+
+/**
+ * Factor out common words in statically defined options.
+ * Create prefix and/or suffix labels.
+ */
+function trimOptions(options: MenuOption[]):
+    {options: MenuOption[]; prefix?: string; suffix?: string;} {
+  let hasImages = false;
+  const trimmedOptions = options.map(([label, value]): MenuOption => {
+    if (typeof label === 'string') {
+      return [parsing.replaceMessageReferences(label), value];
+    }
+
+    hasImages = true;
+    // Copy the image properties so they're not influenced by the original.
+    // NOTE: No need to deep copy since image properties are only 1 level deep.
+    const imageLabel = label.alt !== null ?
+        {...label, alt: parsing.replaceMessageReferences(label.alt)} :
+        {...label};
+    return [imageLabel, value];
+  });
+
+  if (hasImages || options.length < 2) return {options: trimmedOptions};
+
+  const stringOptions = trimmedOptions as [string, string][];
+  const stringLabels = stringOptions.map(([label]) => label);
+
+  const shortest = utilsString.shortestStringLength(stringLabels);
+  const prefixLength = utilsString.commonWordPrefix(stringLabels, shortest);
+  const suffixLength = utilsString.commonWordSuffix(stringLabels, shortest);
+
+  if ((!prefixLength && !suffixLength) ||
+      (
+          // One or more strings will entirely vanish if we proceed.  Abort.
+          shortest <= prefixLength + suffixLength))
+    return {options: stringOptions};
```

Comment:
Will do! I think originally I had the return on line 714, though the formatter probably bumped it to a new line.

Reply:
Fixed!

Reply:
Could you still move the comment so that which part is the predicate is a bit easier to parse? I think either put it right above the if-statement, or right about the return statement.